includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -109,6 +109,8 @@ public static function normalize_url( $url ) {
 			$url = $home . $path;
 		}
 
+		$url = apply_filters( 'amp_normalize_url', $url );
```

Comment:
I think the original URL should also be included in case the default normalization is wrong, no?

Also there seems to be a problem where this existing code could be quite wrong in other environments where HTTPS is required:

https://github.com/ampproject/amp-wp/blob/966771907ce3016c3512e2e729bd374ff35c4899/includes/utils/class-amp-image-dimension-extractor.php#L91-L93

Bringing it all together:

```php
/**
 * Normalizes the given URL.
 *
 * This method ensures the URL has a scheme and, if relative, is prepended the WordPress site URL.
 *
 * @param string $url URL to normalize.
 * @return string Normalized URL.
 */
public static function normalize_url( $url ) {
	if ( empty( $url ) ) {
		return false;
	}

	if ( 0 === strpos( $url, 'data:' ) ) {
		return false;
	}

	$normalized_url = $url;

	if ( 0 === strpos( $url, '//' ) ) {
		$normalized_url = set_url_scheme( $url, 'http' );
	} else {
		$parsed = wp_parse_url( $url );
		if ( ! isset( $parsed['host'] ) ) {
			$path = '';
			if ( isset( $parsed['path'] ) ) {
				$path .= $parsed['path'];
			}
			if ( isset( $parsed['query'] ) ) {
				$path .= '?' . $parsed['query'];
			}
			$home      = home_url();
			$home_path = wp_parse_url( $home, PHP_URL_PATH );
			if ( ! empty( $home_path ) ) {
				$home = substr( $home, 0, - strlen( $home_path ) );
			}
			$normalized_url = $home . $path;
		}
	}

	/**
	 * Apply filters on the normalized image URL for dimension extraction.
	 *
	 * @since 1.1
	 *
	 * @param string $normalized_url Normalized image URL.
	 * @param string $url            Original image URL.
	 */
	$normalized_url = apply_filters( 'amp_normalized_dimension_extractor_image_url', $normalized_url, $url );
	return $normalized_url;
}
```